CMakeLists.txt

```diff
@@ -306,7 +306,9 @@ add_custom_command(
     COMMAND ./openrct2-cli sprite build \"${CMAKE_BINARY_DIR}/g2.dat\" \"${ROOT_DIR}/resources/g2/sprites.json\"
     WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
 )
-add_custom_target(g2 DEPENDS ${PROJECT} g2.dat)
+add_custom_target(g2 DEPENDS ${PROJECT_NAME} g2.dat)
+
+project(openrct2 CXX)
```

Comment:
this already exists on line 22

Reply:
`PROJECT` is not defined in this file; it inherits it from included `CMakeLists` (odd code organisation... but not my doing). It is specifically defined in the included `src/openrct2-ui/CMakeLists.txt`.
What happens when this file is not included? `PROJECT` is empty and that wreaks havoc later; please see the linked issue.

This change makes the `g2` build depend on the last `project` name: if you build with GUI, it should depend on `openrct2`, exactly as before.
Without GUI, it will depend on `openrct2-cli` (the last included sub-`CMakeLists.txt`), which did not seem to yield any problem on my side.

This is a quick fix to what looks like a broader mess of inter-dependencies, but I am no CMake expert, and this needed rework might be seen as broader than the scope of the issue at hand.